bfh/fields.py

```diff
@@ -63,7 +64,10 @@ def __get__(self, instance, cls=None):
         return instance.__dict__.get(self.field_name)
 
     def __set__(self, instance, value):
-        instance.__dict__[self.field_name] = value
+        if value is None:
```

Comment:
I have been thinking about this a lot and I don't think that I agree. I think that this is the correct place to set the default. We don't care about the default unless we are missing a value, and we want the value to equal that default, this does that. On top of that, it makes sure that we create a new instance of our callable, therefore state is not shared across multiple Field instances. I have written a test that proves this. If we were to put that logic in the __get__ then we instantiate a new default callable every time that we get the field_name. This means that it is completely immutable, and we won't be able to update that field_name at any point.

Ex.

The way it is written:
```
        def test_callable():
            return {"wow": 1}

        class Wow(Schema):
            foo = UnicodeField(default=test_callable)

        my1 = Wow(foo=None)
        my2 = Wow(foo=None)

        oops = my1.foo
        oops["wow"] = 2

       my1.foo = {"wow": 2}
       my2.foo = {"wow": 1}
```

if we were to do it in the __get__:

```
       def test_callable():
            return {"wow": 1}

        class Wow(Schema):
            foo = UnicodeField(default=test_callable)

        my1 = Wow(foo=None)
        my2 = Wow(foo=None)

        oops = my1.foo
        oops["wow"] = 2
        
        my1.foo = {"wow": 1}
        my2.foo = {"wow": 1}
```

I think it makes more sense that we can set the default, but then you should be able to update it.